src/ghidra/p_code_extractor/VarnodeSimple.java

```diff
@@ -0,0 +1,42 @@
+import ghidra.program.util.VarnodeContext;
+import ghidra.program.model.pcode.Varnode;
+import ghidra.program.model.lang.Register;
+
+/**
+ * Wrapper class for Varnode
+ * (https://ghidra.re/ghidra_docs/api/ghidra/program/model/pcode/Varnode.html).
+ * 
+ * Varnodes represent registers, stack offset and other values and are used as
+ * operants for
+ * pcode instructions.
+ * This class is used for clean and simple serialization.
+ */
+public class VarnodeSimple {
+    private String addressspace;
+    private String id;
+    private int size;
+
+    public VarnodeSimple(Varnode varnode, VarnodeContext context) {
+        this.size = varnode.getSize();
+        this.addressspace = varnode.getAddress().getAddressSpace().getName();
+        this.id = varnode.getAddress().toString("0x");
+        if (context.getRegister(varnode) != null) {
+            this.id = context.getRegister(varnode).getName();
+        }
+    }
+
+    public VarnodeSimple(Register register) {
+        this.id = register.getName();
+        this.size = (int) register.getBitLength() / 8;
```

Comment:
If I remember correctly, this code should work as intended. Especially if this is just copied from the old code, then it should be correct. But just to make sure: Have you tested it with flag registers?

Reply:
It originates from the old code. Flag registers get a (byte) size of 1.